examples/js/AnimationClipCreator.js

```diff
@@ -0,0 +1,175 @@
+
+	var tracks = [];
+	var frameStep = duration / morphTargets.length;
+
+	for( var i = 0; i < morphTargets.length; i ++ ) {
+
+		var keys = [];
+
+		if( ( i - 1 ) >= 0 ) {
+
+			keys.push( { time: ( i - 1 ) * frameStep, value: 0 } );
+
+		}
+
+		keys.push( { time: i * frameStep, value: 1 } );
+
+		if( ( i + 1 ) <= morphTargets.length ) {
+
+			keys.push( { time: ( i + 1 ) * frameStep, value: 0 } );
+
+		}
+
+		var morphName = morphTargets[i].name;
+		var trackName = '.morphTargetInfluences[' + morphName + ']';
+		var track = new THREE.KeyframeTrack( trackName, keys );
+
+		tracks.push( track );
+	}
+
+	var clip = new THREE.AnimationClip( 'morphAnimation', duration, tracks );
+	//console.log( 'morphAnimationClip', clip );
+
+	return clip;
+};
+
+THREE.AnimationClipCreator.CreateRotationAnimation = function( period, axis ) {
+
+	var keys = [];
+	keys.push( { time: 0, value: 0 } );
+	keys.push( { time: period, value: 360 } );
+
+	axis = axis || 'x';
+	var trackName = '.rotation[' + axis + ']';
+
+	var track = new THREE.KeyframeTrack( trackName, keys );
+
+	var clip = new THREE.AnimationClip( 'rotate.x', 10, [ track ] );
+	//console.log( 'rotateClip', clip );
+
+	return clip;
+};
+
+THREE.AnimationClipCreator.CreateScaleAxisAnimation = function( period, axis ) {
+
+	var keys = [];
+	keys.push( { time: 0, value: 0 } );
+	keys.push( { time: period, value: 360 } );
+
+	axis = axis || 'x';
+	var trackName = '.scale[' + axis + ']';
+
+	var track = new THREE.KeyframeTrack( trackName, keys );
+
+	var clip = new THREE.AnimationClip( 'scale.x', 10, [ track ] );
+	//console.log( 'scaleClip', clip );
+
+	return clip;
+};
+
+THREE.AnimationClipCreator.CreateShakeAnimation = function( duration, shakeScale ) {
+
+	var keys = [];
+
+	for( var i = 0; i < duration * 10; i ++ ) {
+
+		keys.push( { 
+			time: ( i / 10.0 ),
+			value: new THREE.Vector3( Math.random() * 2.0 - 1.0, Math.random() * 2.0 - 1.0, Math.random() * 2.0 - 1.0 ).multiply( shakeScale )
+		} );
+
+	}
+
+	var trackName = '.position';
+
+	var track = new THREE.KeyframeTrack( trackName, keys );
+
+	var clip = new THREE.AnimationClip( 'shake' + duration, duration, [ track ] );
+	//console.log( 'shakeClip', clip );
+
+	return clip;
+};
+
+
+THREE.AnimationClipCreator.CreatePulsationAnimation = function( duration, pulseScale ) {
+
+	var keys = [];
+
+	for( var i = 0; i < duration * 10; i ++ ) {
+
+		var scaleFactor = Math.random() * pulseScale;
+		keys.push( {
+			time: ( i / 10.0 ),
+			value: new THREE.Vector3( scaleFactor, scaleFactor, scaleFactor )
+		} );
+
+	}
+
+	var trackName = '.scale';
+
+	var track = new THREE.KeyframeTrack( trackName, keys );
+
+	var clip = new THREE.AnimationClip( 'scale' + duration, duration, [ track ] );
+	//console.log( 'scaleClip', clip );
+
+	return clip;
+};
+
+
+THREE.AnimationClipCreator.CreateVisibilityAnimation = function( duration ) {
+
+	var keys = [];
+	keys.push( {
+		time: 0,
+		value: true
```

Comment:
I do not quite understand.

I've made a quick stab at create types for the KeyframeTracks, but I am not sure that is what you mean.

There are three complexities in this PR that I am not sure the best way to solve:

The first is how to have polymorphic interpolate/accumulate/set operators.  This functionality is required both by KeyframeTrack (interpolate/set), and AnimationMixer (accumulate/set).  Interpolating quaternions (.slerp) is different than interpolating vectors (.lerp) and is different than interpolating numbers (just use math directly.)  Also setting objects like Quaternion/Vector is different than setting primitives like number.

The second problem is how to handle setting properties on the scene graph.  It is more complex than the set operation described above.  When setting things on the scene graph one still needs to be aware of whether one is setting an object or a primitive, but one also needs to be aware of whether one is setting an array element (see my functions setValue_propertyIndexed/getValue_propertyIndexed).

Lastly, a lot of the complexity of PropertyBinding is because it uses flexible path definitions to identity nodes.  It supports the following types of paths:

```
.propertyName  // e.g. .visible, .rotation, .scale, .castShadow
.propertyName[accessor]  // e.g.  .rotation[x], .scale[z]
nodeName.propertyName  // e.g. myNode.rotation
nodeID.propertyName[accessor] // e.g. eb0e0f09-c2cd-4be4-b567-c3f9a1160a3c.position[y]
.objectName.propertyName // e.g. .material.transparent
.objectName.propertyName[accessor] // e.g. .material.diffuse[r]
.materials[objectIndex].propertyName // e.g. .materials[0].opacity
.bone[Armature.DEF_cog].position // virtual property, bone name reolved to index at bind time
.morphTarget[morphTargetName] // virtual property, bone name reolved to index at bind time
```

The last two are virtual properties because I am allowing for bone names and morph names to be used as an accessor when binding, and then automatically resolved to the numbered indices for run-time efficiency.

I believe that allowing for these types of paths for tracks allows just about anything in ThreeJS to be animated easily as part of one animation clip.